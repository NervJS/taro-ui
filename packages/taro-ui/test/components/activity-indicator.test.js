import React from 'react'
import { mount } from 'enzyme'
import AtActivityIndicator from '../../lib/components/activity-indicator'
import AtLoading from '../../lib/components/loading'

const sizeStyle = {
  width: '32rpx', // TODO: 应该是 rem
  height: '32rpx'
}

const colorStyle = {
  border: '1px solid #13CE66',
  borderColor: '#13CE66 transparent transparent transparent'
}

describe('ActivityIndicator Component', () => {
  it('render initial ActivityIndicator', () => {
    const component = mount(<AtActivityIndicator />)
    expect(component.find('taro-view-core')).toHaveLength(6)
    expect(component.find(AtLoading)).toHaveLength(1)
    expect(component.html()).toMatchSnapshot()
  })

  it('render ActivityIndicator -- props mode ', () => {
    const component = mount(<AtActivityIndicator mode='center' />)
    expect(
      component.find('.at-activity-indicator--center').length
    ).toBeGreaterThanOrEqual(1)
    expect(component.html()).toMatchSnapshot()
  })

  it('render ActivityIndicator -- props size', () => {
    const component = mount(<AtActivityIndicator size={32} />)
    expect(component.prop('size')).toBe(32)
    expect(component.find('.at-loading').first().prop('style')).toEqual(
      sizeStyle
    )
    expect(component.html()).toMatchSnapshot()
  })

  it('render ActivityIndicator -- props color', () => {
    const component = mount(<AtActivityIndicator color='#13CE66' />)
    expect(component.prop('color')).toBe('#13CE66')
    expect(
      component.find('.at-loading__ring').first().prop('style')
    ).toMatchObject(colorStyle)
    expect(component.html()).toMatchSnapshot()
  })

  it('render ActivityIndicator -- props content ', () => {
    const component = mount(<AtActivityIndicator content='加载中' />)
    expect(
      component.find('.at-activity-indicator__content').first().text()
    ).toBe('加载中')
    expect(component.html()).toMatchSnapshot()
  })

  it('render ActivityIndicator -- props isOpened ', () => {
    const component = mount(<AtActivityIndicator isOpened={false} />)
    expect(
      component.find('.at-activity-indicator').length
    ).toBeGreaterThanOrEqual(1)
    expect(component.find('.at-activity-indicator--isopened').length).toBe(0)
    expect(component.html()).toMatchSnapshot()
  })
})
