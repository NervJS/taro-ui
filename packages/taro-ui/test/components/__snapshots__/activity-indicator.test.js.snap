// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityIndicator Component render ActivityIndicator -- props color 1`] = `"<taro-view-core class=\\"at-activity-indicator at-activity-indicator--isopened\\"><taro-view-core class=\\"at-activity-indicator__body\\"><taro-view-core class=\\"at-loading\\"><taro-view-core class=\\"at-loading__ring\\" style=\\"border: 1px solid #13ce66; border-color: #13ce66 transparent transparent transparent;\\"></taro-view-core><taro-view-core class=\\"at-loading__ring\\" style=\\"border: 1px solid #13ce66; border-color: #13ce66 transparent transparent transparent;\\"></taro-view-core><taro-view-core class=\\"at-loading__ring\\" style=\\"border: 1px solid #13ce66; border-color: #13ce66 transparent transparent transparent;\\"></taro-view-core></taro-view-core></taro-view-core></taro-view-core>"`;

exports[`ActivityIndicator Component render ActivityIndicator -- props content  1`] = `"<taro-view-core class=\\"at-activity-indicator at-activity-indicator--isopened\\"><taro-view-core class=\\"at-activity-indicator__body\\"><taro-view-core class=\\"at-loading\\"><taro-view-core class=\\"at-loading__ring\\"></taro-view-core><taro-view-core class=\\"at-loading__ring\\"></taro-view-core><taro-view-core class=\\"at-loading__ring\\"></taro-view-core></taro-view-core></taro-view-core><taro-text-core class=\\"at-activity-indicator__content\\">加载中</taro-text-core></taro-view-core>"`;

exports[`ActivityIndicator Component render ActivityIndicator -- props isOpened  1`] = `"<taro-view-core class=\\"at-activity-indicator\\"><taro-view-core class=\\"at-activity-indicator__body\\"><taro-view-core class=\\"at-loading\\"><taro-view-core class=\\"at-loading__ring\\"></taro-view-core><taro-view-core class=\\"at-loading__ring\\"></taro-view-core><taro-view-core class=\\"at-loading__ring\\"></taro-view-core></taro-view-core></taro-view-core></taro-view-core>"`;

exports[`ActivityIndicator Component render ActivityIndicator -- props mode  1`] = `"<taro-view-core class=\\"at-activity-indicator at-activity-indicator--center at-activity-indicator--isopened\\"><taro-view-core class=\\"at-activity-indicator__body\\"><taro-view-core class=\\"at-loading\\"><taro-view-core class=\\"at-loading__ring\\"></taro-view-core><taro-view-core class=\\"at-loading__ring\\"></taro-view-core><taro-view-core class=\\"at-loading__ring\\"></taro-view-core></taro-view-core></taro-view-core></taro-view-core>"`;

exports[`ActivityIndicator Component render ActivityIndicator -- props size 1`] = `"<taro-view-core class=\\"at-activity-indicator at-activity-indicator--isopened\\"><taro-view-core class=\\"at-activity-indicator__body\\"><taro-view-core class=\\"at-loading\\"><taro-view-core class=\\"at-loading__ring\\"></taro-view-core><taro-view-core class=\\"at-loading__ring\\"></taro-view-core><taro-view-core class=\\"at-loading__ring\\"></taro-view-core></taro-view-core></taro-view-core></taro-view-core>"`;

exports[`ActivityIndicator Component render initial ActivityIndicator 1`] = `"<taro-view-core class=\\"at-activity-indicator at-activity-indicator--isopened\\"><taro-view-core class=\\"at-activity-indicator__body\\"><taro-view-core class=\\"at-loading\\"><taro-view-core class=\\"at-loading__ring\\"></taro-view-core><taro-view-core class=\\"at-loading__ring\\"></taro-view-core><taro-view-core class=\\"at-loading__ring\\"></taro-view-core></taro-view-core></taro-view-core></taro-view-core>"`;
